backend/workflow_manager/endpoint/migrations/0004_allow_multi_source_inputs_for_fs.py

```diff
@@ -0,0 +1,99 @@
+# Generated by Django 4.2.1 on 2024-08-21 02:56
+
+from typing import Any, Optional
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("endpoint", "0003_alter_workflowendpoint_connection_type"),
+    ]
+
+    def migrate_to_support_multiple_folder_input(apps: Any, schema_editor: Any) -> None:
```

Comment:
Is this data migration necessary? Could we handle the input directly in the code instead? For example, we could:

- Split the input into a list if it contains commas, or
- If it's a single string (representing a single directory, as in the existing behavior), wrap it in a list.

Reply:
I believe what he suggested makes sense. We can avoid this migration.

Reply:
> Split the input into a list if it contains commas, or

@muhammad-ali-e in order to accept a list of values, we've made a JSON schema change which returns an array. So it'll not be a single string with commas - we could keep the UI input as a string and enforce as a list, but it might not be the best UX.

> If it's a single string (representing a single directory, as in the existing behavior), wrap it in a list.

This might be possible but we'll eventually end up with a stale code branch that we might not know when to clean up. 
```
if isinstance(folders, str):
    # Convert to a list of 1 item
```

@johnyrahul @hari-kuriakose what are your thoughts on this?

Reply:
MoM of the huddle we had regarding this - agreed to go with migration itself.
- Will not be deleting the old `rootFolder` key though
- Will be filtering for all `SOURCE` endpoints (both `FILESYSTEM` and `API` types)

Will try to take a dump of staging data to test this

Reply:
Agree with a migration. This will avoid any unnecessary logic as the JSON schema is fixed to an array now.